src/helpers/time-range.ts

```
@@ -97,3 +97,14 @@ function combineRanges(range1: TimeRange, range2: TimeRange): TimeRange {
         endTime: (range2.endTime > range1.endTime) ? range2.endTime : range1.endTime
     };
 }
+
+function applyOffsetToDate(dataPoint: DeviceDataPoint | DeviceDataV2Point, datePropertyName: keyof DeviceDataPoint & keyof DeviceDataV2Point) {
+    const dateStr = dataPoint[datePropertyName];
+
+    const dateOffsetPropertyName = (datePropertyName + 'Offset') as keyof DeviceDataV2Point;
+    if (dataPoint.hasOwnProperty(dateOffsetPropertyName)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)